content/blog/aws-cdk-on-pulumi-1.0/index.md

```diff
@@ -0,0 +1,257 @@
+\____/\____/_/ /_/\__, /_/   \__,_/\__/\__,_/_/\__,_/\__/_/\____/_/ /_/____(_)
+                 /____/
+
+
+        Congratulations, your service has successfully deployed on AWS App Runner.
+
+
+
+Open it in your browser at https://2ez3iazupm.us-west-2.awsapprunner.com/
+
+Try the workshop at https://apprunnerworkshop.com
+Read the docs at https://docs.aws.amazon.com/apprunner
+```
+
+
+## Pulumi takes CDK apps to the next level
+
+Combine CDK constructs with any of the features of Pulumi programs to deploy faster, easier and to any cloud. Use Pulumi functions and stack references to connect to pre-existing infrastructure and mix in resources from any provider to bring *all* of your infrastructure under management. 
+
+For example, we can use a Pulumi function to look up AMIs to use when creating an EC2 instance in CDK. Then, we connect a cloudfront distribution over the public IP to provide caching and DDOS protection. 
+
+```typescript
+import * as pulumicdk from '@pulumi/cdk';
+import * as aws from '@pulumi/aws';
+
+const app = new pulumicdk.App('app', (scope: pulumicdk.App): pulumicdk.AppOutputs => {
+    const stack = new pulumicdk.Stack('example-stack');
+    // use getAmiOutput to lookup the AMI instead of ec2.LookupMachineImage
+    const ami = aws.ec2.getAmiOutput({
+        owners: ['amazon'],
+        mostRecent: true,
+        filters: [
+            {
+                name: 'name',
+                values: ['al2023-ami-2023.*.*.*.*-arm64'],
+            },
+        ],
+    });
+
+    const region = aws.config.requireRegion();
+    const machineImage = ec2.MachineImage.genericLinux({
+        [region]: pulumicdk.asString(ami.imageId),
+    });
+
+    const instance = new ec2.Instance(this, 'Instance', {
+        vpc,
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.T4G, ec2.InstanceSize.MICRO),
+        machineImage,
+    });
+
+   // Return the public IP of the instance
+   this.publicIp = this.asOutput(ec2Instance.publicIp);
+});
+
+// Pulumi's abstraction to create a CloudFront distribution
+const distribution = new aws.cloudfront.Distribution("my-cdn-distribution", {
+    enabled: true,
+    origins: [{
+        originId: ec2Instance.urn,
+        domainName: ec2PublicIp.apply(ip => `${app.publicIp}.compute-1.amazonaws.com`), // Constructed domain name
+        customOriginConfig: {
+            originProtocolPolicy: "http-only",
+            httpPort: 80,
+            httpsPort: 80,
+            originSslProtocols: ["TLSv1.2"],
+        },
+    }],
+    defaultCacheBehavior: {
+        targetOriginId: ec2Instance.urn,
+        viewerProtocolPolicy: "allow-all",
+        allowedMethods: ["GET", "HEAD", "OPTIONS"],
+        cachedMethods: ["GET", "HEAD"],
+        // Define cache behavior settings...
+    },
+    // Additional settings like price class, custom error responses, etc.
+    // ...
+    viewerCertificate: {
+        cloudfrontDefaultCertificate: true,
+    },
+});
```

Comment:
Cory, I would appreciate your sanity check on this I smashed some examples together based on our conversation Friday about the most compelling features from a CDK users perspective, but it probably needs some fixes.

Reply:
🤦 I had intended to use Cloudflare, but was moving quickly and asked the AI the wrong question. We do have an example creating a record on cloudflare so I think we could test with that account+domain: [pulumi-cloudflare/examples/record/ts/index.ts at master · pulumi/pulumi-cloudflare](https://github.com/pulumi/pulumi-cloudflare/blob/master/examples/record/ts/index.ts)